Library/Formula/apcupsd.rb

```diff
@@ -0,0 +1,90 @@
+  desc "Daemon for controlling APC UPSes"
+  homepage "http://www.apcupsd.org"
+  url "https://downloads.sourceforge.net/project/apcupsd/apcupsd%20-%20Stable/3.14.13/apcupsd-3.14.13.tar.gz"
+  sha256 "57ecbde01d0448bf8c4dbfe0ad016724ae66ab98adf2de955bf2be553c5d03f9"
+
+  depends_on "libusb-compat"
+  depends_on "gd"
+
+  def install
+    cd "src/apcagent" do
+      # Install apcagent.app to `prefix` instead of /Applications.
+      inreplace "Makefile", "Applications", "#{"#{prefix}".gsub(%r{/^\//}, "")}"
+    end
+
+    sysconfdir = "#{etc}/apcupsd"
+
+    cd "platforms/darwin" do
+      # Install binaries to `prefix`.
+      # Patch submitted to upstream repo:
+      #   https://sourceforge.net/p/apcupsd/mailman/message/34627459/
+      inreplace "Makefile", "/sbin", "$(sbindir)"
+
+      # Install the LaunchDaemon and kernel extension to `prefix`.
+      inreplace "Makefile", "/Library/LaunchDaemons", "#{prefix}/Library/LaunchDaemons"
+      inreplace "Makefile", "/System/Library/Extensions", kext_prefix
+
+      # Tell the launch script and LaunchDaemon to use the appropriate sbin and config directories.
+      inreplace "apcupsd-start", "/sbin", opt_sbin
+      inreplace "apcupsd-start", "/etc/apcupsd", sysconfdir
+      inreplace "org.apcupsd.apcupsd.plist", "/sbin", opt_sbin
+      inreplace "org.apcupsd.apcupsd.plist", "/etc/apcupsd", sysconfdir
+
+      # Custom uninstaller is not needed as everything is installed to `prefix`.
+      inreplace "Makefile", /.*apcupsd-uninstall.*/, ""
+    end
+
+    system "./configure", "--disable-option-checking",
+                          "--prefix=#{prefix}",
+                          "--sbindir=#{sbin}",
+                          "--sysconfdir=#{sysconfdir}",
+                          "--with-cgi-bin=#{sysconfdir}",
+                          "--enable-cgi",
+                          "--enable-usb",
+                          "--enable-modbus-usb"
+
+    system "make", "install"
+  end
+
+  def caveats
+    <<-EOS.undent
+      The apcupsd configuration files are located here:
+        #{etc}/apcupsd
+
+      In order for apcupsd to work, the apcupsd kernel extension
+      must be installed by the root user:
+
+        sudo cp -pR #{kext_prefix}/ApcupsdDummy.kext /System/Library/Extensions/
+        sudo chown -R root:wheel /System/Library/Extensions/ApcupsdDummy.kext
+        sudo touch /System/Library/Extensions/
+
+      #{if MacOS.version >= :el_capitan then <<-EOS.undent
+            Note: On OS X El Capitan, System Integrity Protection (SIP) prevents
+            this kernel extension from loading. It is needed in order for apcupsd
+            to interface with UPSes connected via USB. Without this extension you
+            will have to unplug and plug the USB cable in again after each reboot.
+            This can only be circumvented by partially disabling SIP, which can
+            only be done from the Recovery partition:
+
+              csrutil enable --without kext
```

Comment:
I *really* don't want anything in Homebrew formulae that tells people to even partially disable SIP. I'd rather just include the disclaimer about needing to unplug/plug back in.